wordpress-mobile/WordPress-iOS
# Add support for AT.

Fixes #8967.

The WPKit counter-part is https://github.com/wordpress-mobile/WordPressKit-iOS/pull/30.

After tearing my hair out over how bizarre this somehow behaves, finally, it's ready ✨ 

You can now kick-off the AT process from the iOS app! It feels magical. And horrifying.

So, basically, like AT.

All the user-facing strings are ripped from the Android app :) 

Also inspired by the Android app, I've added a _very_ verbose logging to this — it's inherently a tricky process, so we might see users writing in about it.

Here's a demo video: http://klausa.pl/stuff/at.mov (sorry for not being embeddable!).

To test:
This is a bit nigthmare-ish to test, given what the underlying operations here are.
I have a test site I can give you to play with, but it's sort-of a one-way process (not *really*, but it's not tririval.) 

If you have SA rights on WPCom, easiest way would be to:
1. Create a new site
2. Add Business plan to it
3. Associate a custom domain with it
4. Make sure it's public
4. in the iOS app, try to install a plugin.
5. Verify that the flow executes correctly!

If you don't have SA privileges, ping me. I'll help you out :)

It'd be also worthwhile to verify the pre-flight failure cases. The easiest way to test it is:
1. Go trough steps 1-4 as above
2. In the iOS app, go to plugins and select any random plugin.
3. In Calypso, change the site privacy setting to `private`.
4. Tap "Install" in the iOS app.
5. Verify that the app handles the failure gracefully and displays a toast message.

it should look something like this:
![simulator screen shot - iphone x - 2018-08-24 at 04 39 57](https://user-images.githubusercontent.com/1594081/44562010-caba4b00-a757-11e8-87ed-131eec56ef56.png)
